experiment/template/eVOLVER.py

```diff
@@ -0,0 +1,629 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import time
+import pickle
+import shutil
+import logging
+import argparse
+import numpy as np
+from scipy import stats
+from socketIO_client import SocketIO, BaseNamespace
+
+import custom_script
+from custom_script import EXP_NAME, OD_POWER, PUMP_CAL_FILE
+from custom_script import EVOLVER_IP, EVOLVER_PORT, OPERATION_MODE
+from custom_script import STIR_INITIAL, TEMP_INITIAL
+
+# Should not be changed
+# vials to be considered/excluded should be handled
+# inside the custom functions
+VIALS = [x for x in range(16)]
+
+SAVE_PATH = os.path.dirname(os.path.realpath(__file__))
+EXP_DIR = os.path.join(SAVE_PATH, EXP_NAME)
+
+logger = logging.getLogger('eVOLVER')
+
+EVOLVER_NS = None
+
+class EvolverNamespace(BaseNamespace):
+    start_time = None
+    use_blank = False
+    OD_initial = None
+
+    def on_connect(self, *args):
+        print("Connected to eVOLVER as client")
+        logger.info('connected to eVOLVER as client')
+
+    def on_disconnect(self, *args):
+        print("Disconected from eVOLVER as client")
+        logger.info('disconnected to eVOLVER as client')
+
+    def on_reconnect(self, *args):
+        print("Reconnected to eVOLVER as client")
+        logger.info("reconnected to eVOLVER as client")
+
+    def on_broadcast(self, data):
+        logger.debug('broadcast received')
+        elapsed_time = round((time.time() - self.start_time) / 3600, 4)
+        logger.debug('elapsed time: %.4f hours' % elapsed_time)
+        print("{0}: {1} Hours".format(EXP_NAME, elapsed_time))
+        # are the calibrations in yet?
+        if self.check_for_calibrations():
```

Comment:
Instead of doing a bunch of nested if/else logic with increasing indentations, can you just do the check and `return` after logging the error if you don't want to continue? It gets hard to follow when you're 4 nested logic statements in.